openedx/frontend-app-course-authoring
# feat: [FC-0044] Unit page header section

### Description
This pull request introduces a new Course Unit page that will render all Unit components (XBlocks) and all Course Unit-relative information.

The primary features were implemented:
- Page Breadcrumbs: a navigation system has been implemented to provide users with a clear path back to the course outline page and scroll to particular section and subsection blocks. 
- Page Header Block: block which display Course Unit title and configuration controls: edit - allows changing the page title, configure - for now isn't functional.
- Preview Buttons: A set of two buttons (“View Live Version” and “Preview”) that allow the course creator to view a preview of the course and a live version of the course, respectively.

### Design
https://www.figma.com/file/YeKFwSpyLaJFDs3f3p3TSa/Studio-1%3A1-mock-ups?node-id=599%3A23595

<img width="707" alt="image" src="https://github.com/openedx/frontend-app-course-authoring/assets/17108583/bf90d06b-983d-4ff9-a3ec-dec477f40549">

### Testing instructions
1. Run master devstack.
2. Start platform make dev.up.lms+cms+frontend-app-course-authoring and make checkout on this branch.
3. Enable new Unit page by adding a waffle flad `contentstore.new_studio_mfe.use_new_unit_page` in CMS admin panel.
5. Make sure that MFE setting `ENABLE_UNIT_PAGE=true` is enabled.
6. Go to Course Unit page from the Course Outline page.
